src/auth_service/user_management/claims_repository/core/seed.py

```diff
@@ -105,7 +123,44 @@ async def _add_configured_data_steward_claims(
                 continue
             else:
                 log.warning("Added missing data steward with external ID %r.", ext_id)
-        claim = create_data_steward_claim(user.id)
+        else:
+            _check_data_steward_info(data_steward, user)
+        # add the IVA of the data steward
+        iva_data = IvaBasicData(
+            type=data_steward.iva_type, value=data_steward.iva_value
+        )
+        try:
+            iva = await iva_dao.find_one(
+                mapping={
+                    "user_id": user.id,
+                    "type": iva_data.type,
+                    "value": iva_data.value,
+                }
+            )
+        except MultipleHitsFoundError:
+            log.warning(
+                "IVA %s: %r is not unique for user with external ID %r.",
+                iva_data.type,
+                iva_data.value,
+                ext_id,
+            )
+            continue
```

Comment:
This path only get's triggered if we somehow create the "same" IVA through the API, just at a different time, right?

What about local configuration changes to the IVA, i.e. an the IVA for an already established Data Steward changes in the startup config?
As far as I can see, the old IVA inaccesible to this code path and in contrast to the corresponding claim is not removed. From the new code it looks like we want the IVA from config to be mapped to a claim, but previous from config IVAs could still be around for the corresponding user. So the notion here is that there's a primary IVA set by the current config and the IVA from the old config remains valid?

Reply:
This error could happen if the data steward created two same IVAs via the profile page. The UI and backend should not allow this, but we should still catch the case that there is a duplicate IVA in the database for some reason.

I think you want to suggest that in this case we should better raise an error because this is a serious error condition that should not be just ignored. I have changed this in 9429548f now, let me know if you're ok with that solution.

If a different IVA is specified in the config, then it will be created and the claim associated with the changed IVA. The old IVA will not be removed or invalidated, and I think that's the totally fine. Or do you see a problem with it? If the data steward crates a new 2FA token, then both old and new IVA will be invalidated.

Reply:
Thanks, was just wondering about if behaviour matches intention wrt not removing old IVAs.
Raising also makes more sense imo, would be a bit weird to suddenly have claims revoked due to duplicate IVAs.
If that's unlikely to happen, then that should warrant further investigation instead of just logging it.